src/crypto.c

```diff
@@ -185,7 +185,7 @@ zxerr_t crypto_sign(const hd_path_t path,
                                sizeof(messageDigest),
                                &messageDigestSize));
 
-    if (messageDigestSize != 32) {
+    if (messageDigestSize != CX_SHA256_SIZE) {
```

Comment:
- I think the audit report pointed to replacing the `32` on [this line](https://github.com/vacuumlabs/app-flow/blob/merging-with-onflow-latest-development-branch/src/crypto.c#L135). However, I don't agree with the report suggestion. There should be a new defined constant `CX_SHA3_256_SIZE` (also equal to 32). SHA256 refers to SHA-2, while here we are working with SHA-3

- The line modified in this PR should actually remain 32, or should be replaced by another new constant (the elliptic curve order size), otherwise the check doesn't make sense from a cryptographic perspective. In this specific case, all constants happen to be 32 and the code passes, but if we evolve the code to support more curves/hashing, the code would become incorrect.

Reply:
I can make a suggestion PR in a few hours to clean up the magic numbers from a cryptographic point of a view. It would only improve the code readability, but wouldn't have any functional change on the current code.

Reply:
here is my suggestion PR: https://github.com/vacuumlabs/app-flow/pull/4